pkg/cmd/experimental/diagnostics/cluster.go

```diff
@@ -50,7 +50,7 @@ func (o DiagnosticsOptions) buildClusterDiagnostics(rawConfig *clientcmdapi.Conf
 		case clustdiags.MasterNodeName:
 			diagnostics = append(diagnostics, &clustdiags.MasterNode{KubeClient: kclusterClient, OsClient: clusterClient, ServerUrl: serverUrl, MasterConfigFile: o.MasterConfigLocation})
 		case clustdiags.ClusterRegistryName:
-			diagnostics = append(diagnostics, &clustdiags.ClusterRegistry{KubeClient: kclusterClient, OsClient: clusterClient})
+			diagnostics = append(diagnostics, &clustdiags.ClusterRegistry{KubeClient: kclusterClient, OsClient: clusterClient, PreventModification: o.PreventModification})
```

Comment:
The only other place (currently) is in deciding whether to run DiagnosticPod. I've tried to avoid having diagnostics modify anything, even if it's just to create and delete like these cases, since something could go wrong with the delete and leave junk lying around. However I couldn't find any way around it for these cases. So `--prevent-modification` is there for people who are concerned about this and only want completely read-only diagnostics.